### ISSUE TITLE
[contract v2] Create a listing error from Profile

### ISSUE THREAD

**Issue by jabby09** [KlimaDAO/klimadao#1665]
I currently have 1 tonne of VCS-981-2017

Replicate steps below but other things to note:
- If I start the transaction from my Portfolio (i.e. click SELL on asset), and then complete the Create a listing form, I do start from the APPROVE tab in the modal, however, I am being asked to confirm 2 tonnes
- My suspicion here is that a listing I created using the 5 min expiry preview link and then that listing disappeared (that is one showing for $3.89 in the screenshot below). It looks like the 1 tonne was returned to my wallet but maybe not?? since above I'm being asked to confirm 2 tonnes but I don't actually have a listing or 2 tonnes in my wallet.

![Screenshot 2023-10-13 at 11 07 13 AM](https://github.com/KlimaDAO/klimadao/assets/103535195/7f90ea8a-617f-4ba8-8424-389e32a01c5f)
![Screenshot 2023-10-13 at 11 07 33 AM](https://github.com/KlimaDAO/klimadao/assets/103535195/59a24b96-3602-469d-b92d-118c32195eef)

To replicate:

- From Profile page, click CREATE NEW LISTING
- Select Quantity of 1
- Set Price
- Click Create Listing button
- In the modal, the APPROVE step is skipped
- Click SUBMIT and get an error

![Screenshot 2023-10-13 at 10 56 05 AM](https://github.com/KlimaDAO/klimadao/assets/103535195/5edca51a-c3b9-4b10-91e0-8206f99f17c8)

**Comment by jabby09** [comment]
copy of console

```
index.tsx:144 Error in onAddListing Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x6476edf98Bdc5EE62f9D4E4BF8932d69Ed685483","to":"0x5DC7083cD6Ff9A46C36661750faBcbB9aE2abed2","data":"0x9057f289000000000000000000000000ecf4a1b92a463c843cdcb7cb7a2f2ddfe07651bb0000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000000010eff00000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000000000000000000000000000000000000659ff25a","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":3,"message":"execution reverted","data":"0x72681ef6000000000000000000000000ecf4a1b92a463c843cdcb7cb7a2f2ddfe07651bb0000000000000000000000000000000000000000000000000de0b6b3a76400000000000000000000000000000000000000000000000000000de0b6b3a7640000"},"stack":"{\n  \"code\": -32603,\n  \"message\": \"Internal JSON-RPC error.\",\n  \"data\": {\n    \"code\": 3,\n    \"message\": \"execution reverted\",\n    \"data\": \"0x72681ef6000000000000000000000000ecf4a1b92a463c843cdcb7cb7a2f2ddfe07651bb0000000000000000000000000000000000000000000000000de0b6b3a76400000000000000000000000000000000000000000000000000000de0b6b3a7640000\"\n  },\n  \"stack\": \"Error: Internal JSON-RPC error.\\n    at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:16:10940)\\n    at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:16:13880)\\n    at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:16:14490)\\n    at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:44665)\\n    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:45664\\n    at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-4.js:19:37935\"\n}\n  at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:16:10940)\n  at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:16:13880)\n  at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:16:14490)\n  at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:44665)\n  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:45664\n  at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-4.js:19:37935"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.7.2)
    at h.makeError (index.js:224:1)
    at h.throwError (index.js:233:1)
    at v (json-rpc-provider.js:76:1)
    at c.<anonymous> (json-rpc-provider.js:575:1)
    at Generator.throw (<anonymous>)
    at s (json-rpc-provider.js:6:42)
```

**Comment by Atmosfearful** [comment]
You are correct. If the API response `user.listings` does not contain the expired listing, then the frontend is not subtracting that from the total, and is not preventing you from creating a second listing.

So this will be resolved when #1655 is fixed.
